misc.php

```diff
@@ -902,7 +906,7 @@
 				$all = true;
 				break;
 			}
-			elseif(is_numeric($fid))
+			elseif((int)$fid)
```

Comment:
It cause XSS vulnerability in misc.php.
example: `misc.php?action=syndication&forums[]=1&forums[]=2<script>alert(%27hi%27)</script>&version=rss2.0&limit=15`.